src/nxt_conf_validation.c

```diff
@@ -377,6 +383,16 @@ static nxt_conf_vldt_object_t  nxt_conf_vldt_http_members[] = {
 };
 
 
+static nxt_conf_vldt_object_t  nxt_conf_vldt_router_members[] = {
+    {
+        .name       = nxt_string("threads"),
+        .type       = NXT_CONF_VLDT_INTEGER,
```

Comment:
It seems we need to add more checks to its value. It's similar to "process" with less check.
```
static nxt_int_t
nxt_conf_vldt_processes(nxt_conf_validation_t *vldt, nxt_conf_value_t *value,
    void *data)
{
    int64_t                         int_value;
    nxt_int_t                       ret;
    nxt_conf_vldt_processes_conf_t  proc;

    if (nxt_conf_type(value) == NXT_CONF_INTEGER) {
        int_value = nxt_conf_get_number(value);

        if (int_value < 1) {
            return nxt_conf_vldt_error(vldt, "The \"processes\" number must be "
                                       "equal to or greater than 1.");
        }

        if (int_value > NXT_INT32_T_MAX) {
            return nxt_conf_vldt_error(vldt, "The \"processes\" number must "
                                       "not exceed %d.", NXT_INT32_T_MAX);
        }

        return NXT_OK;
    }
    ...
```

Reply:
I suspect I can just re-use one of the "existing" "threads" check...

Reply:
Correct.

Reply:
Right, this one should do the trick...

```c
static nxt_int_t                                                                
nxt_conf_vldt_threads(nxt_conf_validation_t *vldt, nxt_conf_value_t *value,     
    void *data)                                                                 
{                                                                               
    int64_t  threads;                                                           
                                                                                
    threads = nxt_conf_get_number(value);                                       
                                                                                
    if (threads < 1) {                                                          
        return nxt_conf_vldt_error(vldt, "The \"threads\" number must be "      
                                   "equal to or greater than 1.");              
    }                                                                           
                                                                                
    if (threads > NXT_INT32_T_MAX) {                                            
        return nxt_conf_vldt_error(vldt, "The \"threads\" number must "         
                                   "not exceed %d.", NXT_INT32_T_MAX);          
    }                                                                           
                                                                                
    return NXT_OK;                                                              
} 
```

Reply:
Unless of course we change the name... shame to basically duplicate that function...

Reply:
You could leave it until the team are happy with the usage.

Reply:
Thank you for the team's efforts. I am a developer from China. There are 700M people using nginx products here. if ok I will promote it.

Reply:
Sure, the more the merrier!